h/services/annotation_moderation.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h import models
+
+
+class AnnotationModerationService(object):
+    def __init__(self, session):
+        self.session = session
+
+    def create(self, annotation):
+        """
+        Create an annotation nipa flag.
+
+        This hides the given annotation from anybody except its author and the
+        group moderators.
+
+        In case the given annotation already has a moderation flag, this method
+        is a no-op.
+
+        :param annotation: The annotation to hide from others.
+        :type annotation: h.models.Annotation
+        """
+
+        query = self.session.query(models.AnnotationModeration) \
+                            .filter_by(annotation=annotation)
+
+        if query.count() > 0:
```

Comment:
But an `EXISTS` followed by an `INSERT` is more than fine for this rarely-used endpoint.